### PR TITLE
Standardise package structure

### LINKED ISSUE #18: Standardise package structure and tooling with other BG repos
The tooling is different to other tools we use, and is missing some elements (like a `[dev]` extra). The package structure and tooling can just be taken from [our cookiecutter](https://github.com/neuroinformatics-unit/python-cookiecutter) and looking at other repos.

### PR DESCRIPTION
**Pull request by IgorTatarnikov**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

## Description
The structuring and tooling of this package needs to be standardised with other BrainGlobe packages.

**What is this PR**

- [x] Other

**What does this PR do?**
Standardises this package to match other BrainGlobe packages.

## References
closes #18 

## How has this PR been tested?
Tested locally and on GitHub actions

## Checklist:

- [x] The code has been tested locally
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,7 @@
-# Configuring https://pre-commit.ci/
-ci:
-    autoupdate_schedule: monthly
-
```
Comment (adamltyson) [inline .pre-commit-config.yaml:4]: Why is this removed?

File: pyproject.toml [hunk 2]
```diff
@@ -1,20 +1,130 @@
+[project]
+name = "brainglobe-registration"
+authors = [{name = "Adam Tyson", email= "code@adamltyson.com"}]
```
Comment (adamltyson) [inline pyproject.toml:3]: I don't think I can claim much credit for this repo anymore! For long term support reasons, I wonder if we should start using something like this across all repos?
```suggestion
authors = [{name = "Igor Tatarnikov", email= "hello@brainglobe.info"}]
```

Any thoughts @alessandrofelder?

File: pyproject.toml [hunk 3]
```diff
@@ -1,20 +1,130 @@
...
+readme = "README.md"
+requires-python = ">=3.9.0"
+dynamic = ["version"]
+
+license = {text = "BSD-3-Clause"}
+
+classifiers = [
+    "Development Status :: 2 - Pre-Alpha",
+    "Framework :: napari",
+    "Intended Audience :: Developers",
```
Comment (adamltyson) [inline pyproject.toml:14]:
```suggestion
    "Intended Audience :: Science/Research",
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,20 +1,130 @@
...
+    "qtpy",
+    "itk-elastix",
+    "pytransform3d"
+]
+
+[project.urls]
+"Homepage" = "https://github.com/brainglobe/brainglobe-registration"
+"Bug Tracker" = "https://github.com/brainglobe/brainglobe-registration/issues"
+"Documentation" = "https://github.com/brainglobe/brainglobe-registration#README.md"
+
```
Comment (adamltyson) [inline pyproject.toml:38]:
```suggestion
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,20 +1,130 @@
...
+"Source Code" = "https://github.com/brainglobe/brainglobe-registration"
+"User Support" = "https://github.com/brainglobe/brainglobe-registration/issues"
```
Comment (adamltyson) [inline pyproject.toml:40]:
```suggestion
"User Support" = "https://forum.image.sc/tag/brainglobe"
```

File: pyproject.toml [hunk 6]
```diff
@@ -1,20 +1,130 @@
...
+    "napari>=0.4.18",
+    "bg-atlasapi",
+    "numpy",
+    "qtpy",
+    "itk-elastix",
+    "pytransform3d"
+]
+
+[project.urls]
+"Homepage" = "https://github.com/brainglobe/brainglobe-registration"
```
Comment (adamltyson) [inline pyproject.toml:35]:
```suggestion
"Homepage" = "https://brainglobe.info"
```

File: pyproject.toml [hunk 7]
```diff
@@ -1,20 +1,130 @@
...
+exclude = ["__init__.py","build",".eggs"]
+select = ["I", "E", "F"]
+fix = true
+
+[tool.cibuildwheel]
+build = "cp39-* cp310- cp311-*"
+
+[tool.cibuildwheel.macos]
+archs = ["x86_64", "arm64"]
+
```
Comment (adamltyson) [inline pyproject.toml:112]: Not needed as this is pure Python.
```suggestion
```

**Review by adamltyson — APPROVED** [review]
LGTM. Some tiny suggestions.

File: pyproject.toml (hunk 2 above)
Comment (alessandrofelder) [reply pyproject.toml:3]: Yea, or even something like
```
authors = [{name = "Brainglobe Developers", email= "hello@brainglobe.info"}]
```
Or even "Brainglobe Core Developers"?

Comment (alessandrofelder) [reply pyproject.toml:3]: Especially on repos where we all contribute significantly... this one is definitely by and large just Igor so far!

Comment (adamltyson) [reply pyproject.toml:3]: Yeah I dunno. The single author entry doesn't really tally with how OSS is developed. OTOH, I want to be able to promote individuals as much as possible.

Comment (adamltyson) [reply pyproject.toml:3]: although nobody reads this, so maybe BG developers is ok.

Comment (alessandrofelder) [reply pyproject.toml:3]: Made this a discussion for the [next dev meeting](https://brainglobe.zulipchat.com/#narrow/stream/414089-developer-meeting/topic/developer-meeting-07-12-2023/near/404686128)!

Comment (alessandrofelder) [reply pyproject.toml:3]: > OTOH, I want to be able to promote individuals as much as possible. 

Agree - I wonder how other OSS projects do this :thinking: maybe a yearly updated list of who was an official BG dev in the last year?

Comment (adamltyson) [reply pyproject.toml:3]: [numpy use the generic "NumPy Developers"](https://github.com/numpy/numpy/blob/c09c89cdf6f139e7fc473cd05b46f78bfbb8b1b8/pyproject.toml#L17)

Comment (IgorTatarnikov) [reply pyproject.toml:3]: I'm partial to:
```suggestion
authors = [{name = "Brainglobe Developers", email= "hello@brainglobe.info"}]
```

Comment (adamltyson) [reply pyproject.toml:3]: Fine by me.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (adamltyson) [reply .pre-commit-config.yaml:4]: Yeah I think it's useful. I've raised a PR to add this to the cookiecutter - https://github.com/neuroinformatics-unit/python-cookiecutter/pull/93
